lib/core/src/irods_default_paths.cpp

```diff
@@ -18,31 +21,83 @@ namespace irods {
         try {
             boost::filesystem::path path{dl_info.dli_fname};
             path = boost::filesystem::canonical(path);
-            path.remove_filename().remove_filename().remove_filename(); // Removes filename and the two directories (usr and lib) between libirods_common.so and base of irods install
+            path.remove_filename();
             return path;
         } catch(const boost::filesystem::filesystem_error& e) {
             THROW(-1, e.what());
         }
     }
 
+    boost::filesystem::path
+    get_irods_root_directory() {
+        try {
+            boost::filesystem::path path = get_irods_lib_directory();
```

Comment:
If these functions return `boost::filesystem::path`'s, you can replace the explicit types with `auto` and let the compiler figure it out. Totally up to you.
```c++
auto path = get_irods_lib_directory();
```